vector/types.go

```diff
@@ -0,0 +1,123 @@
+package vector
+
+import (
+	"net/netip"
+
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/pkg/nano"
+
+	"github.com/RoaringBitmap/roaring"
+)
+
+type any interface {
```

Comment:
`any` is a built-in alias for `interface{}`. Is there another name that might make sense here? Maybe `materializable`?